pkg/state/impl/etcd/etcd_test.go

```diff
@@ -0,0 +1,97 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package etcd_test
+
+import (
+	"context"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/protobuf"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/cosi-project/runtime/pkg/state/conformance"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"go.etcd.io/etcd/server/v3/embed"
+	"go.etcd.io/etcd/server/v3/etcdserver/api/v3client"
+
+	"github.com/cosi-project/state-etcd/pkg/state/impl/etcd"
+)
+
+func TestInterfaces(t *testing.T) {
+	t.Parallel()
+
+	assert.Implements(t, (*state.CoreState)(nil), new(etcd.State))
+}
+
+func TestEtcdConformance(t *testing.T) {
+	t.Parallel()
+
+	err := protobuf.RegisterResource(conformance.PathResourceType, &conformance.PathResource{})
+	if err != nil {
+		t.Fatalf("failed to register resource: %v", err)
+	}
+
+	withEtcd(t, func(s state.State) {
+		suite.Run(t, &conformance.StateSuite{
+			State:      s,
+			Namespaces: []resource.Namespace{"default"},
+		})
+	})
+}
+
+func withEtcd(t *testing.T, f func(state.State)) {
+	tempDir := t.TempDir()
+
+	cfg := embed.NewConfig()
+	cfg.Dir = tempDir
+
+	peerURL, err := url.Parse("http://localhost:0")
+	if err != nil {
+		t.Fatalf("failed to parse URL: %v", err)
+	}
+
+	clientURL, err := url.Parse("http://localhost:0")
+	if err != nil {
+		t.Fatalf("failed to parse URL: %v", err)
+	}
+
+	cfg.LPUrls = []url.URL{*peerURL}
+	cfg.LCUrls = []url.URL{*clientURL}
+
+	e, err := embed.StartEtcd(cfg)
```

Comment:
my understanding is that we don't need to make `etcd` listen actually, as embedded etcd object implements client interfaces (see my other comment about client interfaces)

Reply:
Hm, I am not sure how to make it not listen to any port. `embed.StartEtcd()` seems to be the only "proper" way to get the `embed.Etcd` object and it wants its URLs to be set, otherwise defaulting to 2379&2380.

Reply:
got it, interesting!